### ISSUE TITLE
[Bug] keras.ops.conv / depthwise_conv / separable_conv skip input-channel validation in symbolic path

### ISSUE THREAD

**Issue by rstar327** [keras-team/keras#22852]
### Summary

`keras.ops.conv`, `keras.ops.depthwise_conv`, and `keras.ops.separable_conv` validate that the input tensor's channel dimension matches the kernel's input channels in the **eager** path, but the **symbolic** path silently returns an output shape based on the (mismatched) kernel — masking the bug until eager execution.

This matches the pattern of #22515 (separable_conv shape) and #22472 (rgb_to_hsv channel validation), where eager and symbolic paths disagree on validation.

### Reproduction

```python
import os
os.environ["KERAS_BACKEND"] = "tensorflow"
import keras
import keras.ops as ops
import numpy as np

# 1) keras.ops.conv
x = np.random.normal(size=(1, 10, 3)).astype("float32")        # 3 input channels
kernel = np.random.normal(size=(3, 5, 4)).astype("float32")     # kernel expects 5 input channels (mismatch)

try:
    print("eager   :", ops.conv(x, kernel, padding="valid").shape)
except Exception as e:
    print("eager   : raises", type(e).__name__)

print("symbolic:", ops.conv(keras.Input(shape=(10, 3)), kernel, padding="valid").shape)

# 2) keras.ops.depthwise_conv
dw = np.random.normal(size=(3, 5, 1)).astype("float32")        # depthwise kernel expects 5 input channels
try:
    print("eager   :", ops.depthwise_conv(x, dw, padding="valid").shape)
except Exception as e:
    print("eager   : raises", type(e).__name__)

print("symbolic:", ops.depthwise_conv(keras.Input(shape=(10, 3)), dw, padding="valid").shape)

# 3) keras.ops.separable_conv
pw = np.random.normal(size=(1, 5, 4)).astype("float32")
try:
    print("eager   :", ops.separable_conv(x, dw, pw, padding="valid").shape)
except Exception as e:
    print("eager   : raises", type(e).__name__)

print("symbolic:", ops.separable_conv(keras.Input(shape=(10, 3)), dw, pw, padding="valid").shape)
```

### Observed

```
# conv
eager   : raises ValueError
symbolic: (None, 8, 4)

# depthwise_conv
eager   : raises ValueError ("input channels=3, kernel input channels=5")
symbolic: (None, 8, 5)

# separable_conv
eager   : raises ValueError
symbolic: (None, 8, 4)
```

### Expected

Both paths should raise the same `ValueError` when input channels don't match the kernel's input channels. The symbolic path currently fabricates an output shape that the model would never actually be able to compute.

### Root cause

In `keras/src/ops/nn.py`:

- `Conv.compute_output_spec` (~line 1510) computes output shape from `kernel.shape[-1]` and the spatial dims, but does not check `inputs.shape[channel_dim] == kernel.shape[-2]`.
- `DepthwiseConv.compute_output_spec` (~line 1553) has the same gap.
- `SeparableConv.compute_output_spec` (~line 1659) inherits the gap by delegating to `depthwise_conv`.

The eager path validates via `keras/src/backend/common/backend_utils.check_depthwise_conv_input_channels` (added in #22687), but that helper isn't reused in the symbolic path.

### Impact

Models built from a `keras.Input` placeholder with wrong-channel kernels are accepted at construction time and only fail at eager execution, when the user has already finished wiring up the model. A clear early `ValueError` at shape inference is consistent with how other ops validate.

**Comment by rahulrathnavel** [comment]
@maitry63 Hi, I’m interested in working on this issue. Are you currently working on it,or would it be okay if I take it?
